### DISCUSSION TITLE
How to help

### DISCUSSION THREAD

**Discussion by coolrazor007** [ballle98/AqualinkD#66] [Ideas]
Balle98, first of all, thank you SOOO much for continuing to support this project and your hard work in getting it to work for PDA systems.  I have been off and on using it with my PDA pool.

So, I'm a DevOps guy and trying to bolster my GitHub/portfolio and help projects however I can.  Since I already use this software I'd like to see if there's a way I can help.  I'm a novice coder (relevant languages: python, bash, yaml), but I am decent at Docker, K8s, Terraform, Jenkins and linux.

Would it make sense to run AqualinkD in a container?  Totally doable except the USB part.  Not sure how to approach that, but if we solved that then AqualinkD could run on a lot more hardware and "install" would possibly be easier.

Also, I have found myself in situations where we (read: my kids) lost the PDA remote and would love to figure out a way to incorporate the aquaweb project into this project.  I realize you have previously mentioned that AqualinkD is much smarter but that shouldn't be a hindrance right?  Even as basic as a script that switches between running the AqualinkD process and aquaweb process would be possible (pretty low-tech I realize, but it'd work).

Basically, how can I help?  I'd love to be able to be able to create a CI/CD pipeline for this but we don't have a virtual pool controller simulator to test against do we?  Maybe unittest instead?

**Comment by coolrazor007** [comment]
Spoke a little too soon, looks like you can pair the USB to a container: https://stackoverflow.com/questions/24225647/docker-a-way-to-give-access-to-a-host-usb-or-serial-device

**Comment by ballle98** [comment]
@coolrazor007 not sure if putting this in a container would add much value.  You don't need to scale out this app.  Might mitigate some of the security issues but still need to fix them.  There are several open issues that need work.
